common/serializer.go

```diff
@@ -26,8 +26,13 @@ func Unmarshal(metadata *HttpMetadata, responseMapping interface{}) error {
 		return nil
 	}
 
-	if err := json.Unmarshal(metadata.ResponseBody, &responseMapping); err != nil {
-		return err
+	switch metadata.Headers.Header.Get("Content-Type") {
+	case "text/csv":
```

Comment:
This is probably wrong in some cases, because we could also get `text/csv charset=utf-8`
So either the conditiional should be `starts with` or `contains`